### PR TITLE
[Prometheus] Update changelog and minor cleanup.

### PR DESCRIPTION
**Pull request by Yun-Ting**
## Changes
Update changelog and minor cleanup.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md [hunk 1]
```diff
@@ -2,10 +2,12 @@
...
 ## Unreleased
 
-* Split up Prometheus projects based on its hosting mechanism, HttpListener and AspNetCore,
-  into their own projects and assemblies. The shared code for both hosting mechanism
-  now lives in the `OpenTelemetry.Exporter.Prometheus.Shared` project and will not
-  be released.
+* Split up Prometheus projects based on its hosting mechanism,
+  HttpListener and AspNetCore, into their own projects
+  and assemblies.
+  ([#3512](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3512))
```
Comment (utpilla) [inline src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md:8]: You don't need to add this PR in the CHANGELOG.

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md [hunk 2]
```diff
@@ -2,10 +2,11 @@
...
-* Split up Prometheus projects based on its hosting mechanism, HttpListener and AspNetCore,
-  into their own projects and assemblies. The shared code for both hosting mechanism
-  now lives in the `OpenTelemetry.Exporter.Prometheus.Shared` project and will not
-  be released.
+* Split up Prometheus projects based on its hosting mechanism,
+  HttpListener and AspNetCore, into their own projects
+  and assemblies.
+  ([#3507](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3507))
+  ([#3503](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3503))
   ([#3430](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3430))
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md:10]: This doesn't seem correct (it's inconsistent with other changelog entries), would you double check?

**Review by reyang — CHANGES_REQUESTED** [review]
Style inconsistency.

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md [hunk 3]
```diff
@@ -2,11 +2,12 @@
...
-* Split up Prometheus projects based on its hosting mechanism, HttpListener and AspNetCore,
-  into their own projects and assemblies. The shared code for both hosting mechanism
-  now lives in the `OpenTelemetry.Exporter.Prometheus.Shared` project and will not
-  be released.
-  ([#3430](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3430))
+* Split up Prometheus projects based on its hosting mechanism,
+  HttpListener and AspNetCore, into their own projects
+  and assemblies.
+  ([#3507](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3507),
+  [#3503](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3503) &
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md:9]: `&` might not be a common case (maybe it's one-off in some existing places).

I think the most commonly used approach https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/CHANGELOG.md#120-rc3. So it'll look exactly like `(#3430 #3503 #3507)`.

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md (hunk 3 above)
Comment (reyang) [reply src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md]: It looks like ascending order commonly used in this repo.

**Pull request merged**